### DISCUSSION TITLE
Database size doubled after migration

### DISCUSSION THREAD

**Discussion by palu3492** [sqlalchemy/alembic#951] [Usage Questions] [answered]
Hello, I'm migrating a sqlite3 database with alembic and using render_as_batch so I can alter the type of columns in my tables. This works well but after migration, the database is using double the disk space it was using before the migration. The data in the tables look correct so I'm wondering who is using all this disk space? This is making some of my databases grow by gigabytes. Any ideas on how to solve this?

**Comment by zzzeek** [comment, marked as answer]
not familiar with this.  batch migration renames an existing table and then copies it to a new one.  the original table should be dropped.  perhaps sqlite3 datafiles remain large after tables were dropped?  take a look at [vacuum](https://www.sqlite.org/lang_vacuum.html), this likely will fix it.

**Reply by palu3492** [reply]
That was it! I had not heard of vacuum before but that does the trick. Thanks!
